site-src/geps/gep-1867.md

```diff
@@ -0,0 +1,137 @@
+# GEP-1867: Per-Gateway Infrastructure
+
+* Status: Provisional
+
+## Overview
+
+Gateways represent a piece of infrastructure, whether this is  implemented by cloud load balancers, in-cluster deployments, or other mechanisms.
+These often need vendor-configuration outside the scope of existing APIs.
+For example, the "size" or "version" of the infrastructure to provision.
+
+Today, this is configurable via `GatewayClass.spec.parametersRef`, which offers the ability to attach arbitrary configuration
+to a GatewayClass.
+
+This GEP will explain why that is not sufficient to meet common use cases, and introduce a new field - `infrastructure` - to address these cases.
+
+Related discussions:
+* [Support cluster-local Gateways](https://github.com/kubernetes-sigs/gateway-api/discussions/1247)
+* [Scaling Gateway Resources](https://github.com/kubernetes-sigs/gateway-api/discussions/1355)
+* [Manual deployments](https://github.com/kubernetes-sigs/gateway-api/issues/1687)
+* [Merging Gateways](https://github.com/kubernetes-sigs/gateway-api/pull/1863)
+* [In Cluster Gateway Deployments])(https://github.com/kubernetes-sigs/gateway-api/pull/1757)
+
+## Goals
+
+* Provide the ability to configure arbitrary (implementation specific) attributes about a specific Gateway.
+* Provide the ability to configure a standardized set attributes about a specific Gateway.
+
+## Challenges with GatewayClass
+
+`GatewayClass.spec.parametersRef` is the existing mechanism to configure arbitrary fields on a Gateway.
+However, this introduces operational challenges when configuring Gateways.
+
+### Scope
+
+GatewayClass is a cluster scoped resource. This effectively limits customization to either:
+1. Cluster admin pre-provisions a few classes, lower-privilege Gateway admins select them.
+2. Gateway admin and Cluster admin are the same.
+
+Case 1 works great when there are only a handful of customizations needed.
+For example, you could have a "public" vs "private" class, or a "big" and "small" class.
+This doesn't scale beyond simple cases, however; even the example above requires 4 classes and exponentially grows as more use cases are needed.
+
+Case 2 works fine, but greatly limits use cases, such as [Support namespace scoped implementations](https://github.com/kubernetes-sigs/gateway-api/issues/567).
+Rather than having 1 big ingress for the entire cluster, it is common for clusters to have multiple smaller gateways ([example](https://istio.io/latest/docs/setup/additional-setup/gateway/#gateway-deployment-topologies).
+This can be A gateway per app, per namespace,  per "tenant", or any mix of these.
```

Comment:
I'm on board with you that there are challenges of doing this stuff at the `GatewayClass` level and this "scope" argument here is one of the most compelling to me. It seems to me like we're saying a little bit too much about it though, perhaps we can simplify:

```suggestion
### Scope

As a `Gateway` manager (with RBAC permissions to a specific `Gateway`) I should be able to declaratively make changes to that `Gateway` without the need for access to cluster-scoped resources (`GatewayClass`) and without affecting other `Gateways` managed by the same `GatewayClass`.
```